pygmt/src/binstats.py

```diff
@@ -0,0 +1,123 @@
+    b="binary",
+    h="header",
+    i="incols",
+    r="registration",
+)
+@kwargs_to_strings(I="sequence", R="sequence", i="sequence_comma")
+def binstats(data, **kwargs):
+    r"""
+    Bin spatial data and determine statistics per bin.
+
+    Reads arbitrarily located (x,y[,z][,w]) points
+    (2-4 columns) from ``data`` and for each
+    node in the specified grid layout determines which points are
+    within the given radius.  These point are then used in the
+    calculation of the specified statistic. The results may be
+    presented as is or may be normalized by the circle area to
+    perhaps give density estimates.  Alternatively, select
+    hexagonal tiling instead of a rectangular grid layout.
+
+    Full option list at :gmt-docs:`gmtbinstats.html`
+
+    {aliases}
+
+    Parameters
+    ----------
+    data : str or {table-like}
+        A file name of an ASCII data table or a 2D
+        {table-classes}.
+    outgrid : str or None
+        The name of the output netCDF file with extension .nc to store the grid
+        in.
+    statistic : str
+        **a**\|\ **d**\|\ **g**\|\ **i**\|\ **l**\|\ **L**\|\ **m**\|\ **n**
+        \|\ **o**\|\ **p**\|\ **q**\ [*quant*]\|\ **r**\|\ **s**\|\ **u**
+        \|\ **U**\|\ **z**.
+        Choose the statistic that will be computed per node based on the
+        points that are within *radius* distance of the node.  Select one of
+        **a** for mean (average), **d** for median absolute deviation (MAD),
+        **g** for full (max-min) range, **i** for 25-75% interquartile range,
+        **l** for minimum (low), **L** for minimum of positive values only,
+        **m** for median, **n** the number of values, **o** for LMS scale,
+        **p** for mode (maximum likelihood), **q** for selected quantile
+        (append desired quantile in 0-100% range [50]), **r** for the r.m.s.,
+        **s** for standard deviation, **u** for maximum (upper),
+        **U** for maximum of negative values only, or **z** for the sum.
+    empty : float or int
+        Set the value assigned to empty nodes [Default is NaN].
+    normalize : bool
+        Normalize the resulting grid values by the area represented by the
+        search *radius* [no normalization].
+    search_radius : float or str
+        Sets the *search_radius* that determines which data points are
+        considered close to a node. Append the distance unit.
+        Not compatible with ``tiling``.
+    tiling : str
+        Instead of circular, possibly overlapping areas, select
+        non-overlapping tiling.  Choose between **r**\ ectangular
+        **h**\ exagonal binning. For **r**, set bin sizes via ``spacing`` and
+        the computed statistics are written to the ``outgrid``. For **h**,  a
+        table with the centers of the hexagons and the computed statistics is
+        written to the ouput.  Here, ``spacing`` sets the *y* increment only
+        and the *x*-increment is computed given the geometry.
```

Comment:
> For **h**, a table with the centers of the hexagons and the computed statistics is written to the ouput.

By default, `binstats` generates a grid, but when `tiling="h"`, it outputs some information to an ASCII file instead. Because it behaves differently, I think it would be better NOT to add the alias `tiling` in this PR.